arch/src/x86_64/mod.rs

```diff
@@ -834,6 +836,8 @@ pub fn configure_vcpu(
 
     if let Some(t) = topology {
         update_cpuid_topology(&mut cpuid, t.0, t.1, t.2, cpu_vendor, id);
+    } else {
+        update_cpuid_topology(&mut cpuid, 1, max_vcpus, 1, cpu_vendor, id);
```

Comment:
Surely this should be `boot_vcpus` not `max_vcpus`?

Reply:
Fixed.